frontend/packages/noobaa-storage-plugin/src/components/capacity-card/capacity-card-body.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+
+import { ChartDonut, ChartThemeVariant, ChartThemeColor } from '@patternfly/react-charts';
+import { ChartLineIcon } from '@patternfly/react-icons';
+import { DataPoint } from '@console/internal/components/graphs';
+import { EmptyState, EmptyStateVariant, EmptyStateIcon, Title } from '@patternfly/react-core';
+import { LoadingInline, humanizeBinaryBytesWithoutB } from '@console/internal/components/utils';
+
+// as per the noobaa spec,
+// a. omit 'Others' label
+// b. sort it in descending order
+// c. collect the first 'N' items
+// d. rest of the miscs data should be aggregated
+//    into 'Others' label,
+// and separating Data and Legend array from DataPoint
+const getMetricDataAndLegend = (
```

Comment:
@cloudbehl , I'v tried to reduce the complexity of this function as much as I could. Please take a look.